src/docfx/lib/git/GitUtility.cs

```diff
@@ -254,6 +254,7 @@ private static async Task CloneOrUpdate(string path, string url, IEnumerable<str
             catch (InvalidOperationException ex) when (committishes.Any(rev => ex.Message.Contains(rev)))
             {
                 // Fallback to fetch all branches and tags if the input committish is not supported by fetch
+                depth = depthOne ? "--unshallow" : "";
```

Comment:
make it full clone/fetch again? so the depth==1 only works for refName and tags?

Reply:
`depth == 1` only works for branches and tags, you cannot fetch a single with commit hash with git AFAIK.